ownCloud Intents/GetDirectoryListingIntentHandler.swift

```diff
@@ -25,24 +25,43 @@ import ownCloudAppShared
 typealias GetDirectoryListingCompletionHandler = (GetDirectoryListingIntentResponse) -> Void
 
 @available(iOS 13.0, *)
-public class GetDirectoryListingIntentHandler: NSObject, GetDirectoryListingIntentHandling, OCQueryDelegate {
+public class GetDirectoryListingIntentHandler: NSObject, GetDirectoryListingIntentHandling, OCQueryDelegate, OCCoreDelegate {
+	weak var core : OCCore?
+	var completionHandler : GetDirectoryListingCompletionHandler?
 
-	var completion : GetDirectoryListingCompletionHandler?
+	func complete(with response: GetDirectoryListingIntentResponse) {
+		if let completionHandler = completionHandler {
+			self.completionHandler = nil
 
-	func resolvePath(for intent: GetDirectoryListingIntent, with completion: @escaping (INStringResolutionResult) -> Void) {
+			if let bookmark = core?.bookmark {
+				core = nil
 
+				OCCoreManager.shared.returnCore(for: bookmark, completionHandler: {
+					completionHandler(response)
+				})
+			} else {
+				completionHandler(response)
+			}
+		}
+	}
+
+	func resolvePath(for intent: GetDirectoryListingIntent, with completion: @escaping (INStringResolutionResult) -> Void) {
 		if let path = intent.path {
 			completion(INStringResolutionResult.success(with: path))
 		} else {
 			completion(INStringResolutionResult.needsValue())
 		}
-
 	}
 
 	func provideAccountOptions(for intent: GetDirectoryListingIntent, with completion: @escaping ([Account]?, Error?) -> Void) {
 		completion(OCBookmarkManager.shared.accountList, nil)
 	}
 
+	@available(iOSApplicationExtension 14.0, *)
```

Comment:
In my opinion this code is available since iOS 13.0

Reply:
At least in the headers, `INObjectCollection` is marked as available only as of iOS 14:
<img width="766" alt="Bildschirmfoto 2021-03-29 um 23 04 35" src="https://user-images.githubusercontent.com/639669/112900001-2359dd80-90e3-11eb-83dd-294750326723.png">